docs/developers/Backend.md

```diff
@@ -0,0 +1,125 @@
+## Cromwell Backend Development
+
+### What’s a Backend?
+
+Cromwell is a workflow execution service, which means it takes a representation of work to do (tasks) as well as the
+dependencies and code flow between them. When it comes to actually executing the specific task, Cromwell delegates that
+work to a “backend”. A backend is therefore responsible for the actual execution of a single task on some underlying
+computing platform, such as Google Cloud Platform, Azure, AWS, TES, SLURM, etc. Backends are implemented as a software
+layer between the Cromwell engine and that underlying platform.
+
+The underlying platform services requests for a job to run while the backend shim provides the interface layer between
+Cromwell and the platform. In general, the more sophisticated the platform the thinner the shim needs to be and vice
+versa.
+
+A job from Cromwell’s perspective is a collection of the following information:
+
+* A unix command line to run
+* A mapping of where its input files currently live to where the command line expects them to be
+* A mapping of where the command line will write its outputs to where they should eventually wind up
+* An optional Docker image which if supplied will be the environment in which the command will be run
+* A collection of arbitrary key/value pairs which can be used by the execution platform to tune the request, e.g. amount
+  of memory or the number of CPUs.
+
+The Docker image is not required for a backend but is highly recommended. The bioinformatics workflow field is rapidly
+moving towards a Docker model so one will find better support for their backend if it is built around using Docker
+containers.
+
+The input/output file mappings will depend on the needs of the platform. In the simplest example these values would be
+identical, living on a shared filesystem. However an example of where this would be useful would be a model where an
+input file lives in a cloud bucket and is directly copied onto the machine running the command line followed by copying
+the program’s outputs back to a cloud bucket.
+
+The underlying platform can be extremely simple or have arbitrary levels of complexity as long as one can map from the
+above concepts to running a command. That mapping could happen completely in the platform, the Cromwell backend, or a
+mixture of the two. The implementer of a backend will need to strike the balance which works best for both their needs
+and the particulars of the platform itself.
+
+Throughout the rest of this document the following three terms are used, the difference between them is subtle but
+important:
```

Comment:
Could you add "workflow" for completeness? When reading through the document, the first term related to a unit of work that I came across after this section is "workflow" which made me wonder if I missed something in this section.